### PR TITLE
Add DLLogger support to exp_manager

### PR DESCRIPTION
**Pull request by milesial**
# What does this PR do ?

Adds an option to log with DLLogger in the experiment manager.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Changed exp_manager.py to add a new logger
- Updated the exp_manager documentation

# Usage
In the exp_manager config:

```yaml
enable_dllogger_logger: True
dllogger_logger_kwargs:
    stdout: False
    verbose: False
    json_file: "./dllogger.json"
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [x] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/utils/exp_manager.py [hunk 1]
```diff
@@ -68,6 +68,52 @@ class CheckpointMisconfigurationError(NeMoBaseException):
     """ Raised when a mismatch between trainer.callbacks and exp_manager occurs"""
 
 
+try:
+    import dllogger
+    from dllogger import Verbosity
+
+    class DLLogger(Logger):
```
Comment (titu1994) [inline nemo/utils/exp_manager.py:75]: Please define it in some utility class of nemo.utils or nemo.core.utils and then import here.

Comment (titu1994) [inline nemo/utils/exp_manager.py:75]: Don't put entire class in try catch. We have pattern of try catched import guard, add a boolean flag, and then at init of class, you check the Boolean and raise error if it's not installed. This is for deferred dynamic import checking

**Review by titu1994 — COMMENTED** [review]
Minor changes required. @ericharper for final review. It looks good to me

File: nemo/utils/exp_manager.py (hunk 1 above)
Comment (ericharper) [reply nemo/utils/exp_manager.py:75]: about creating a file here: https://github.com/NVIDIA/NeMo/tree/main/nemo/utils?

Comment (ericharper) [reply nemo/utils/exp_manager.py:75]: Here is an example of an import guard in NeMo: https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/nlp/parts/nlp_overrides.py#L44

**Review by ericharper — CHANGES_REQUESTED** [review]
Overall looks great. Please address @titu1994 's comments and we can merge.

File: nemo/utils/exp_manager.py (hunk 1 above)
Comment (milesial) [reply nemo/utils/exp_manager.py]: Done

Comment (milesial) [reply nemo/utils/exp_manager.py]: Done

**Review by ericharper — DISMISSED** [review]
LGTM. Thanks!

**Comment by milesial** [comment]
Thanks. Addressed the unused import issue. DCO check is failing now, is it because of your merge commit?

**Review by titu1994 — APPROVED** [review]
Looks great

**Pull request merged**